be/src/exec/chunks_sorter_full_sort.cpp

```diff
@@ -40,29 +56,54 @@ Status ChunksSorterFullSort::update(RuntimeState* state, const ChunkPtr& chunk)
 // Accumulate unsorted input chunks into a larger chunk
 Status ChunksSorterFullSort::_merge_unsorted(RuntimeState* state, const ChunkPtr& chunk) {
     SCOPED_TIMER(_build_timer);
+    _staging_unsorted_chunks.push_back(std::move(chunk));
+    _staging_unsorted_rows += chunk->num_rows();
+    _staging_unsorted_bytes += chunk->bytes_usage();
+    return Status::OK();
+}
 
-    if (_unsorted_chunk == nullptr) {
-        // TODO: optimize the copy
-        _unsorted_chunk.reset(chunk->clone_unique().release());
-    } else {
-        _unsorted_chunk->append(*chunk);
+template <typename BinaryColumnType>
+static void reserve_memory(Column* dst_col, const std::vector<ChunkPtr>& src_chunks, size_t col_idx) {
+    auto* binary_dst_col = down_cast<BinaryColumnType*>(dst_col);
+    size_t total_num_bytes = 0;
+    for (const auto& src_chk : src_chunks) {
+        const auto* src_data_col = ColumnHelper::get_data_column(src_chk->get_column_by_index(col_idx).get());
+        const auto* src_binary_col = down_cast<const BinaryColumnType*>(src_data_col);
+        total_num_bytes += src_binary_col->get_bytes().size();
     }
-
-    return Status::OK();
+    binary_dst_col->get_bytes().reserve(total_num_bytes);
 }
 
+static void concat_chunks(ChunkPtr& dst_chunk, const std::vector<ChunkPtr>& src_chunks, size_t num_rows) {
+    DCHECK(!src_chunks.empty());
+    dst_chunk = src_chunks.front()->clone_empty(num_rows);
+    const auto num_columns = dst_chunk->num_columns();
+    for (auto i = 0; i < num_columns; ++i) {
+        auto dst_col = dst_chunk->get_column_by_index(i);
+        auto* dst_data_col = ColumnHelper::get_data_column(dst_col.get());
+        if (dst_data_col->is_binary()) {
+            reserve_memory<BinaryColumn>(dst_data_col, src_chunks, i);
+        } else if (dst_col->is_large_binary()) {
+            reserve_memory<LargeBinaryColumn>(dst_data_col, src_chunks, i);
+        }
```

Comment:
large binary column only used when dataskew is extremely drastic, for an example, a PipelineDriver process 4 billion rows, it may happen in product environment extremely rarely, if it really happens, LargeBinaryColumn can solve these problem.